qos/evm/check.go

```diff
@@ -24,11 +27,18 @@ var _ gateway.QoSEndpointCheckGenerator = &EndpointStore{}
 // TODO_IMPROVE(@commoddity): implement QoS check expiry functionality and use protocol.EndpointAddr
 // to filter out checks for any endpoint which has acurrently valid QoS data point.
 func (es *EndpointStore) GetRequiredQualityChecks(_ protocol.EndpointAddr) []gateway.RequestQoSContext {
-	return []gateway.RequestQoSContext{
+	qualityChecks := []gateway.RequestQoSContext{
```

Comment:
I find it a bit weird that this is called `qualityChecks` but we return a list of `QoSContexts`.

Not modifying because I want to see if this was intentional before modifying it.

Reply:
This is a good call out, but the required refactor may over extend the scope of this PR.

- Quality checks should also be determined by the ServiceState (which has access to EndpointStore)
- I suggest adding a TODO_MVP/TODO_TECHDEBT (will most likely be addressed by the refactor work).

Reply:
Agreed; additionally I also have a (new) pending PR to implement expiry functionality for QoS https://github.com/buildwithgrove/path/pull/200 (which will likely change as there are a few PRs preceding it). This PR updates the checks functionality as well so when we get to it we can revisit this issue.